src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java

```diff
@@ -1,42 +1,225 @@
+    private static final long serialVersionUID = 8561874934736620072L;
     private static final Logger LOG = LogManager.getLogger();
 
     protected BatchInputFileService batchInputFileService;
     protected BatchInputFileType jaggaerUploadFileType;
     protected FileStorageService fileStorageService;
-
+    protected CUMarshalService cuMarshalService;
+    protected ParameterService parameterService;
+    protected ConfigurationService configurationService;
+    
+    @Override
+    protected Client getClient() {
+        return super.getClient(CuMultiPartWriter.class);
+    }
+    
+    @Override
     public void uploadSupplierXMLFiles() {
-        List<String> fileNamesToUpload = batchInputFileService.listInputFileNamesWithDoneFile(jaggaerUploadFileType);
-        LOG.info("uploadSupplierXMLFiles, found {} files to upload", fileNamesToUpload.size());
+        List<String> jaggaerUploadXmlFileNames = batchInputFileService
+                .listInputFileNamesWithDoneFile(jaggaerUploadFileType);
+        LOG.info("uploadSupplierXMLFiles, found {} files to upload", jaggaerUploadXmlFileNames.size());
+        
+        List<JaggaerUploadFileResultsDTO> resultsList = new ArrayList<JaggaerUploadFileResultsDTO>();
+
+        for (String jaggaerXmlFileName : jaggaerUploadXmlFileNames) {
+            if (shouldUploadFilesToJaggaer()) {
+                LOG.info("uploadSupplierXMLFiles. processing {}", jaggaerXmlFileName);
+                JaggaerUploadFileResultsDTO results = uploadJaggaerXmlFile(jaggaerXmlFileName);
+                resultsList.add(results);
+            } else {
+                LOG.info("uploadSupplierXMLFiles. uploading to Jaggaer is turned off, just remove the DONE file for {}",
+                        jaggaerXmlFileName);
+            }
+        }
+        
+        if (shouldUploadFilesToJaggaer()) {
+            generateResultsReport(resultsList);
+        }
+        
+        fileStorageService.removeDoneFiles(jaggaerUploadXmlFileNames);
+    }
+
+    protected boolean shouldUploadFilesToJaggaer() {
+        return getParameterValueBoolean(CUPurapParameterConstants.JAGGAER_ENABLE_UPLOAD_FILES);
+    }
+    
+    private JaggaerUploadFileResultsDTO uploadJaggaerXmlFile(String jaggaerXmlFileName) {
+        JaggaerUploadFileResultsDTO results = new JaggaerUploadFileResultsDTO();
+        results.setFileName(jaggaerXmlFileName);
+        
+        boolean successfulCall = false;
+        int numberOfAttempts = 1;
+        
+        while (!successfulCall && numberOfAttempts < getMaximumNumberOfRetries()) {
+            Client client = getClient();
+            WebTarget target = client.target(buildSupplierUploadURI());
+            Invocation.Builder requestBuilder = target.request();
+            
+            disableRequestChunkingIfNecessary(client, requestBuilder);
+            
+            Response response = requestBuilder.accept(MediaType.APPLICATION_XML)
```

Comment:
You may want to close the `Response` object when you're done with it. The `CURestClientUtils` class has a convenience method for closing it quietly, if it's not possible or practical to use it in a try-with-resources block.